iped-app/src/main/java/iped/app/bootstrap/Bootstrap.java

```diff
@@ -73,7 +73,27 @@ protected void run(String args[]) {
         List<String> finalArgs = new ArrayList<>();
         for (String arg : args) {
             if (arg.startsWith("-Xms") || arg.startsWith("-Xmx")) {
-                heapArgs.add(arg);
+                StringBuffer argStr = new StringBuffer();
+                int i=4;
+                for(;i<arg.length();i++) {
+                    if(Character.isDigit(arg.charAt(i))){
+                        argStr.append(arg.charAt(i));
+                    }else {
+                        break;
+                    }
+                }
+                long parmSize = Integer.parseInt(argStr.toString());
+                switch (arg.charAt(i)) {
+                    case 'T':parmSize*=1024;
+                    case 'G':parmSize*=1024;
+                    case 'M':parmSize*=1024;
+                    case 'K':parmSize*=1024;                        
+                        break;
+                    default:
+                        break;
+                }
+                long memSize = Math.min(parmSize,((com.sun.management.OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean()).getTotalPhysicalMemorySize());
+                heapArgs.add("-Xmx"+memSize/1024+"K");
```

Comment:
I think `-Xms` need to be handled here too, right?

Reply:
I don't know if we should treat -Xms in the same way. If some user pass this parameter intentionally, he wants to make some performance tunning, and for that he should know the installed memory and tune this parameter accordingly, and if wrongly tuned in a way that leads to OOM or IPED halting, he should "feel" the effect and correct it by himself.

BTW, I will include some code to inform that the change in -Xmx has occured in App LOG.

Reply:
I think that we could change the -Xmx to half the installed memory only if it is greater than the total memory, so the user have a little "space" to increase by himself this value. What do you think?

Reply:
> I don't know if we should treat -Xms in the same way. If some user pass this parameter intentionally, he wants to make some performance tunning, and for that he should know the installed memory and tune this parameter accordingly, and if wrongly tuned in a way that leads to OOM or IPED halting, he should "feel" the effect and correct it by himself.

The problem is that in line 75 `-Xms` is already handled, so the current code would transform a `-Xms##` into a `-Xmx##`, which I guess is not the desired behavior, right?

About changing or not `-Xms`, the proposed idea is to change a `-Xmx`, even if it was passed by the user, if it exceeds the total physical memory, right? Wouldn't the same reasoning apply to `-Xms`?

Reply:
About the same reasoning: The best option, I think, would be no to change the user chosen passed value via cmd line and let him evaluate the effects. 
But we need to change somewhere the default max heap memory. The "exe" loads with "-Xmx30G", and this value should change. Maybe we could change in this code. But I didn't find where is it. @lfcnassif , could we point out where this value of 30G is specified?

Reply:
> The "exe" loads with "-Xmx30G", and this value should change. Maybe we could change the code that declares this "-Xmx30G". But I didn't find where is it. @lfcnassif , could we point out where this value of 30G is specified?

It is hard coded into the EXE, I'll remove it and leave it blank (no heap option).

About current changes, my proposal is to use half the physical memory as Xmx if it is not specified by user. If it is specified and if it is greater than physical memory, cut it to be equal to physical memory. Seems reasonable?

Reply:
And even if Xms/Xmx are specified by user, I think we can limit them to physical memory, allocating more makes no sense to me. That's why I tagged the issue as bug. Like current EXE, users may create and distribute scripts to open multicases and specify fixed heap values which may be bad depending on the runtime machine, so I think limiting to physical memory is a good approach to workaround user mistakes.